flexmeasures/data/models/reporting/tests/test_tibber_reporter.py

```diff
@@ -0,0 +1,267 @@
+import pytest
+
+from flexmeasures.data.models.reporting.pandas_reporter import PandasReporter
+from flexmeasures.data.models.time_series import Sensor, DataSource, TimedBelief
+from flexmeasures.data.models.generic_assets import GenericAssetType, GenericAsset
+
+import pandas as pd
+from datetime import datetime, timedelta
+from pytz import utc
+
+index = pd.date_range(
+    datetime(2023, 4, 13), datetime(2023, 4, 13, 23), freq="H", tz=utc
+)
+
+entsoe_prices = [
+    97.23,
+    85.09,
+    79.49,
+    72.86,
+    71.12,
+    82.50,
+    102.06,
+    115.04,
+    122.15,
+    105.39,
+    83.40,
+    34.90,
+    -4.50,
+    -50.00,
+    -50.07,
+    -50.00,
+    -0.90,
+    108.10,
+    128.10,
+    151.00,
+    155.20,
+    152.00,
+    134.04,
+    120.10,
+]  # EUR / MWh
+
+tibber_app_price = [
+    29.2,
+    27.7,
+    27.0,
+    26.2,
+    26.0,
+    27.4,
+    29.8,
+    31.3,
+    32.2,
+    30.2,
+    27.5,
+    21.6,  # originally 21.7 due to a rounding error from 216.4569 EUR/MWh (probably via 216.5) to 217 EUR/MWh
+    16.9,
+    11.4,
+    11.4,
+    11.4,
+    17.3,
+    30.5,
+    32.9,
+    35.7,
+    36.2,
+    35.8,
+    33.6,
+    32.0,
+]  # cents/kWh
+
+
+class TibberReporter(PandasReporter):
+    def __init__(self, sensor) -> None:
+        """This class calculates the price of energy of a tariff indexed to the Day Ahead prices.
+        Energy Price = (1 + VAT) x ( EnergyTax + Tiber + DA Prices)
+        """
+
+        # search the sensors
+        EnergyTax = Sensor.query.filter(Sensor.name == "EnergyTax").one_or_none()
+        VAT = Sensor.query.filter(Sensor.name == "VAT").one_or_none()
+        tibber_tariff = Sensor.query.filter(
+            Sensor.name == "Tibber Tariff"
+        ).one_or_none()
+
+        da_prices = Sensor.query.filter(Sensor.name == "DA prices").one_or_none()
+
+        # create the PandasReporter reporter config
+        reporter_config = dict(
+            input_resolution="PT1H",
+            tb_query_config=[
+                dict(sensor=EnergyTax.id, alias="energy_tax_df"),
+                dict(sensor=VAT.id),
+                dict(sensor=tibber_tariff.id),
+                dict(sensor=da_prices.id),
+            ],
+            transformations=[
+                dict(
+                    df_input="sensor_1",
+                    df_output="VAT",
+                    method="droplevel",
+                    args=[[1, 2, 3]],
+                ),
```

Comment:
I'm interested in finding out what we can do to decrease the verbosity of these transformation lists. Specifically, I think we'll be using this exact transformation *a lot*. Here's my idea: we subclass `PandasReporter` into a `SimplePandasReporter` (working title) that automatically applies this transformation on any input DataFrame.

Could be a follow-up ticket of course.

Reply:
Sure, good idea.

Definitely, some simple base reporters might be handy for the day-to-day work, some ideas:
- `SimplePandasReporter`: drops levels other than event_value
- `AggregateReporter`: Aggregate (min, max, mean, sum, ...) the data of all the sensors that are input.
- `EnergyBillReporter`: Computes the price of energy of a given sensor using the energy tariff, tolls, taxes, levies, etc... 

I opened an ticket (Issue #661) to discuss this further.